Added tests for cover API functions

itbel4onok opened the pull request:
Added tests for cover API functions

itbel4onok committed:
Added tests for cover API functions

nikolay-977 commented on src/test/java/actions/OrderAction.java:
```diff
@@ -0,0 +1,36 @@
+package actions;
```
Необходимо исправить: В пакете src.**test**.java должны быть пакеты классов с кейсами. Остальные пакеты, такие как actions, constants и т.д. следует располагать в src.**main**.java.

nikolay-977 commented on src/test/java/CreateOrderNoAuthTest.java:
```diff
@@ -0,0 +1,46 @@
+        response.then()
+                .assertThat().body("success", equalTo(true))
+                .assertThat().body("order.number", notNullValue())
+                .and().statusCode(SC_OK);
```
Необходимо исправить порядок проверки - сначала проверяем statusCode, затем остальное. Иначе в случае если сервер недоступен мы получим NullPointerException, а не status сode 500-й.

nikolay-977 commented on src/test/java/CreateUserEmptyFieldTest.java:
```diff
@@ -0,0 +1,48 @@
+        this.fieldValue = fieldValue;
+    }
+
+    @Parameterized.Parameters(name = "field: {0}, value: {1}")
```
Отлично: в параметризованных тестах для аннотации `@Parameterized.Parameters` применяется аргумент name

nikolay-977 commented on src/test/java/actions/OrderAction.java:
```diff
@@ -0,0 +1,36 @@
+import static io.restassured.RestAssured.given;
+
+public class OrderAction extends BaseApi {
+    public OrderAction(){
```
Можно улучшить: для расстановки пробелов и табуляции в соответствии с настройками Code Style в Idea используй сочетание клавиш Ctrl+Alt+L. Также можно установить курсор на корневую директорию и нажать клавиш Ctrl+Alt+L, отметить чек-боксы: cleanup code, а также optimize imports(для оптимизации импортов) и rearrange entries(для изменения порядка переменных и методов в соответствии с best practics)

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из фалйа README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

itbel4onok committed:
Updated README file

itbel4onok committed:
- Update style by Idea code style
- Updated file locations;
- Changing order of verifications for response
- Updated README file

itbel4onok committed:
Merge branch 'main' into develop2
